### DISCUSSION TITLE
关于 ImmortalWrt 未利用的 eMMC 空间

### DISCUSSION THREAD

**Discussion by cherinyy** [immortalwrt/immortalwrt#1000] [Q&A] [answered]
我在 [ImmortalWrt Firmware Selector](https://firmware-selector.immortalwrt.org/) 下载了 FastRhino R68S `23.05-SNAPSHOT` 版本的 SquashFS 固件，我发现 ImmortalWrt 仅使用了 300+ MiB 的 eMMC 空间，导致剩下的 14 GiB 空间的浪费。

我目前手动格式化了未利用的空间，创建了一个新的分区，用于存储一些想要持久化保存的数据，但是我担忧一个问题：在 LuCI 界面通过`备份与升级`升级固件，是否会导致该分区的丢失？我甚至不清楚 FastRhino R68S 是否能够通过 LuCI 界面更新固件？

如果会，那么 ImmortalWrt 能否在构建时就创建第三个分区来利用剩下的空间？

**Comment by 1715173329** [comment, marked as answer]
在不修改固件原本分区的前提下，自行手动创建的分区在升级时不会受影响。

**Reply by cherinyy** [reply]
好的，那么官方构建时能否就创建这一分区呢？我注意到它似乎无法完成自动挂载，即便我在 `admin/system/mounts` 启用了该挂载点。
不知道是不是跟我使用 `Btrfs` 有关系。

**Reply by 1715173329** [reply]
> 好的，那么官方构建时能否就创建这一分区呢？

不能，但你可以使用 imagebuilder 自行修改默认 rootfs 大小。由于目前 asu 限制，修改默认 rootfs 大小后无法使用值守式升级。

我不推荐你使用 btrfs，这文件系统本身就是种灾难，openwrt 上更是灵车。传统 ext4 已足够。

**Reply by jh0072110** [reply]
歪个楼，就1XG空间你要拿来干什么…这种大小现如今不就是用来…浪费的么……

**Reply by cherinyy** [reply]
真把软路由当 AIO 用啦？这点空间你放持久化配置文件、日志，Docker 镜像文件不行吗？
